stdlib/pqueue.mli

```diff
@@ -0,0 +1,208 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                       Jean-Christophe Filliâtre                        *)
+(*                                                                        *)
+(*   Copyright 2023 CNRS                                                  *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Priority queues.
+
+    The {!Pqueue} module implements a data structure of priority queues,
+    given a totally ordered type for priorities. This is a mutable
+    data structure. Smaller values are considered to have a higher
+    priority (i.e., these are min-priority queues).
+
+    The implementation uses a heap stored in a dynamic array, and is
+    therefore reasonably efficient: accessing the minimum element
+    takes constant time, and insertion and removal take time
+    logarithmic in the size of the priority queue. Note that
+    [of_array] runs in linear time (and thus must be preferred to
+    repeated insertions with [add]).
+
+    It is fine to have several elements with the same priority.
+    It is guaranteed that the element returned by [min_elt] (or
+    [min_elt_opt]) is the one that is removed from the priority queue
+    by [pop_min] (or [pop_min_opt] or [remove_min]). This is important
+    in many algorithms, (e.g. when peeking at several priority queues
+    and then selecting one to remove from).
+
+    @since 5.2
```

Comment:
I believe it's too late for 5.2 (see d46f79c898bf52e3bf76027d4570754916c617e5), it should be 5.3 instead. I'll let @octachron confirm.

Reply:
Indeed, the change would be in 5.3 not 5.2 .

Reply:
changed to 5.3